tests/integration/test_hedged_requests/test.py

```diff
@@ -0,0 +1,70 @@
+import os
+import sys
+import time
+
+import pytest
+
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+from helpers.cluster import ClickHouseCluster
+from helpers.network import PartitionManager
+
+cluster = ClickHouseCluster(__file__)
+
+# Cluster with 1 shard of 2 replicas. node is the instance with Distributed table.
+node = cluster.add_instance(
+    'node', with_zookeeper=True, main_configs=['configs/remote_servers.xml'], user_configs=['configs/users.xml'])
+node_1 = cluster.add_instance('node_1', with_zookeeper=True, stay_alive=True, user_configs=['configs/users1.xml'])
+node_2 = cluster.add_instance('node_2', with_zookeeper=True)
+
+config = '''<yandex>
+    <profiles>
+        <default>
+            <{setting}>30</{setting}>
+        </default>
+    </profiles>
+</yandex>'''
+
+
+@pytest.fixture(scope="module")
+def started_cluster():
+    try:
+        cluster.start()
+
+        node_1.query('''CREATE TABLE replicated (id UInt32, date Date) ENGINE =
+            ReplicatedMergeTree('/clickhouse/tables/replicated', 'node_1')  ORDER BY id PARTITION BY toYYYYMM(date)''')
+
+        node_2.query('''CREATE TABLE replicated (id UInt32, date Date) ENGINE =
+            ReplicatedMergeTree('/clickhouse/tables/replicated', 'node_2')  ORDER BY id PARTITION BY toYYYYMM(date)''')
+       
+        node.query('''CREATE TABLE distributed (id UInt32, date Date) ENGINE =
+            Distributed('test_cluster', 'default', 'replicated')''')
+
+        yield cluster
+
+    finally:
+        cluster.shutdown()
+
+def process_test(sleep_setting_name, receive_timeout_name):
+    node_1.replace_config('/etc/clickhouse-server/users.d/users1.xml', config.format(setting=sleep_setting_name))
+
+    # Restart node to make new config relevant
+    node_1.restart_clickhouse(30)
+    
+    # Without hedged requests select query will last more than 30 seconds,
+    # with hedged requests it will last just around 1-2 second
+
+    start = time.time()
+    node.query("SELECT * FROM distributed");
+    query_time = time.time() - start
+    
+    print(query_time)
```

Comment:
I suppose test should check that query_time, e.g., < 10 or something

Reply:
Also, more tests are needed. At least, with `max_parallel_replicas > 1`.
Also, test when all replicas are stale (but query still should be executed).
Also, that it is possible to use single replica for several for several offsets, when `max_parallel_replicas > 1`, but only one replica is not stale.

Reply:
> Also, test when all replicas are stale (but query still should be executed).
> Also, that it is possible to use single replica for several for several offsets, when max_parallel_replicas > 1, but only one replica is not stale.

I thought we have already have such tests. Or you mean to combine such situations with replicas sleeping?